modules/electrophysiology_browser/php/models/electrophysioevents.class.inc

```diff
@@ -57,7 +78,340 @@ class ElectrophysioEvents
+
+        $factory = \NDB_Factory::singleton();
+        $user    = $factory->user();
+        $db      = $factory->database();
+
+        if ($user->hasPermission('electrophysiology_browser_edit_annotations')) {
+
+            //If no derivative files exist, must create new files
+            $eventFileID = $db->pselect(
+                "SELECT EventFileID
+                      FROM physiological_task_event
+                      WHERE PhysiologicalFileID=:PFID",
+                ['PFID' => $this->_physioFileID]
+            );
+
+            //Get data from POST request
+            $metadata = [
+                'Description' => $metadata['description'],
+                'Sources'     => $metadata['sources'],
+                'Author'      => $metadata['author']
+            ];
+            error_log(json_encode($metadata));
+
+            if (is_null($instance_id)) {
+                // TODO: Support Instance INSERT
+                $instance_id = 1;
+            }
+
+            $instance = [
+                'Onset'    => $instance_data['onset'],
+                'Duration' => $instance_data['duration'],
+            ];
+
+            // TODO: Support Event Instance Insert
+            if (!empty($eventFileID)) {
+                // Update physiological_task_event
+                $db->update(
+                    'physiological_task_event',
+                    $instance,
+                    ['PhysiologicalTaskEventID' => $instance_id]
+                );
+
+                $db->update(
+                    'physiological_event_file',
+                    ['LastUpdate' => date("Y-m-d H:i:s")],
+                    ['PhysiologicalFileID' => $this->_physioFileID]
+                );
+            }
+
+            $taskEvent = $db->pselect(
+                'SELECT * FROM physiological_task_event
+            WHERE PhysiologicalTaskEventID=:PTEID',
+                ['PTEID' => $instance_id]
+            );
+
+            $extraColumns = $db->pselect(
+                'SELECT opt.*
+                  FROM physiological_task_event_opt AS opt
+                  WHERE opt.PhysiologicalTaskEventID=:PTEID',
+                ['PTEID' => $instance_id]
+            );
+
+            return [
+                'instance'      => $taskEvent[0],
+                'extra_columns' => $extraColumns,
+            ];
+        }
+        return [];
+    }
+
+    /**
+     * Deletes one event instance
+     *
+     * @param int $physiologicalTaskEventID PhysiologicalTaskEventID
+     *
+     * @return void
+     */
+    function deleteEvent(int $physiologicalTaskEventID): void
+    {
+        $db = \NDB_Factory::singleton()->database();
+
+        $physioFileID = $db->pselectone(
+            'SELECT PhysiologicalFileID
+              FROM physiological_task_event
+              WHERE PhysiologicalTaskEventID=:taskEventID',
+            ['taskEventID' => $physiologicalTaskEventID]
+        );
+
+        // TODO: Check that this cascades properly to rel tables
```

Comment:
Does it? Else, it is missing some delete.

Reply:
Comment removed. This is related to the `physiological_task_event_opt` and `physiological_task_event_hed_rel` tables with `physiological_task_event` FKs